lab-davidw/routes/lure-routes.js

```diff
@@ -0,0 +1,129 @@
+'use strict';
+
+const debug = require('debug')('http:lure-routes');
+const FishingLure = require('../model/fishingLure');
+const storage = require('../lib/storage');
+
+module.exports = function(router) {
+  router.get('/api/lure', function(req, res) {
+    debug('GET /api/lure');
+    if(req.url.query.id) {
+      storage.fetchItem('toy', req.url.query.id)
+      .then(lure => {
+        res.writeHead(200, {'Content-Type': 'application/json'});
+        res.write(JSON.stringify(lure));
+        res.end();
+      })
+      .catch(err => {
+        console.error(err);
+        res.writeHead(404, {'Content-Type': 'text/plain'});
+        res.write('not found');
+        res.end();
+      });
+      return;
+    }
+
+    res.writeHead(400, {'Content-Type': 'text/plain'});
+    res.write('bad request');
+    res.end();
+  });
+
```

Comment:
getting your POST request to work is key. for the other CRUD operations to work, first something has to be created by the POST, then u can check to see if it exists with a GET, you can update it with a PUT, or delete with the DELETE.